gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/GravatarQuickEditorActivity.kt

```diff
@@ -0,0 +1,142 @@
+package com.gravatar.quickeditor.ui
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.Build.VERSION.SDK_INT
+import android.os.Build.VERSION_CODES.TIRAMISU
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.activity.compose.setContent
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.appcompat.app.AppCompatActivity
+import com.gravatar.quickeditor.ui.GravatarQuickEditorActivity.GravatarEditorActivityArguments
+import com.gravatar.quickeditor.ui.editor.AuthenticationMethod
+import com.gravatar.quickeditor.ui.editor.GravatarQuickEditorParams
+import com.gravatar.quickeditor.ui.editor.bottomsheet.GravatarQuickEditorBottomSheet
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Activity that hosts the [GravatarQuickEditorBottomSheet] composable.
+ * This activity is used to show the Gravatar Quick Editor in a bottom sheet.
+ *
+ * @see GravatarEditorActivityArguments
+ */
+public class GravatarQuickEditorActivity : AppCompatActivity() {
+    private var avatarHasChanged: Boolean = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            overrideActivityTransition(OVERRIDE_TRANSITION_OPEN, 0, 0)
+        } else {
+            @Suppress("DEPRECATION")
+            overridePendingTransition(0, 0)
+        }
+
+        val arguments = requireNotNull(intent.parcelable<GravatarEditorActivityArguments>(EXTRA_QE_ARGUMENTS))
+        setContent {
+            GravatarQuickEditorBottomSheet(
+                gravatarQuickEditorParams = arguments.gravatarQuickEditorParams,
+                authenticationMethod = arguments.authenticationMethod,
+                onAvatarSelected = { avatarHasChanged = true },
+                onDismiss = { finishWithResult() },
+            )
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putBoolean("avatarHasChanged", avatarHasChanged)
```

Comment:
⛏️ Can we extract `avatarHasChanged` to a property?

Reply:
Sure! [02dd869](https://github.com/Automattic/Gravatar-SDK-Android/pull/425/commits/02dd869b162f3eed0400189f75a3282ca1e448b6)